src/stk/image/volume.h

```diff
@@ -190,6 +196,7 @@ namespace stk
         float3 _origin; // Origin in world coordinates
         float3 _spacing; // Size of a voxel
         Matrix3x3f _direction; // Cosine directions of the axes
+        Matrix3x3f _inverse_direction;
```

Comment:
We should probably refactor the metadata-bit so we can put the direction there as well. I assume direction won't be used that much.

Reply:
Well, the direction matrix (or its inverse) is used every time we access a point (e.g. inside the cost functions), so that's probably quite a lot of usage. I fear keeping it inside a dictionary may hurt performance. Unless I misunderstood what you mean.

Reply:
Yeah, should probably not have it in a dictionary, but we could have a something like this
 
```
struct MetaData {
    Matrix3x3f direction;
    Dict variables;
};
```

Reply:
That may work better. But I am not sure about this, the direction is not really meta-data, at least not more than origin and spacing, it actually defines the geometry of the image. I don't think it is very consistent to put it in a different place than origin and spacing, so maybe we should move all three of them in such struct.

Reply:
Yeah, that makes sense.